### PR TITLE
[releng] Migration 2021-06

### PR DESCRIPTION
**Pull request by arnauddieumegard**
This PR contains many changes on different aspects of Kitalpha:
- Releng: migration of the TP (covers step 1 in #677)
- Massactions modifications
- Product configuration modifications (.product)
- Build modifications in pom files

### CODE REVIEW

File: .project [hunk 1]
```diff
@@ -5,7 +5,13 @@
 	<projects>
 	</projects>
 	<buildSpec>
+		<buildCommand>
+			<name>net.sf.eclipsecs.core.CheckstyleBuilder</name>
+			<arguments>
+			</arguments>
```
Comment (bzendagui) [inline .project:11]: test

File: .project [hunk 2]
```diff
@@ -5,7 +5,13 @@
 	<projects>
```
Comment (bzendagui) [inline .project:5]: It this file important? if not, can you remove it please?

File: massactions/tests/plugins/org.polarsys.kitalpha.massactions.junit.tests/tests/org/polarsys/kitalpha/massactions/tests/ju/test/view/ColumnHideShowTest.java [hunk 3]
```diff
@@ -90,18 +91,16 @@ private void testColumnHideShow(MAView view) {
     ColumnHideShowLayer columnHideShowLayer = bodyLayer.getColumnHideShowLayer();
 
     // the internal representation of hidden columns
-    Set<Integer> beforeHiddenColumns = Collections.emptySet();
-    Set<Integer> afterHiddenColumns = Collections.emptySet();
+    Set<Integer> beforeHiddenColumns;
+    Set<Integer> afterHiddenColumns;
```
Comment (bzendagui) [inline massactions/tests/plugins/org.polarsys.kitalpha.massactions.junit.tests/tests/org/polarsys/kitalpha/massactions/tests/ju/test/view/ColumnHideShowTest.java:95]: To make the code easier to read and understund, can you please remove the declaration of the two sets from here and do them in the lines 118 and 124 (when the instances are created).

**Review by bzendagui — CHANGES_REQUESTED** [review]
Can you please check all comment, there are some suggestions that we can discuss.

File: massactions/plugins/org.polarsys.kitalpha.massactions.core/src/org/polarsys/kitalpha/massactions/core/table/layer/body/MAColumnReorderLayer.java [hunk 4]
```diff
@@ -31,41 +29,42 @@
...
-  @Override
-  public void handleLayerEvent(ILayerEvent event) {
-    List<Integer> prevColumnIndexOrder = Collections.emptyList();
+    /**
+     * By default the NatTable implementation resets the currently ordered columns whenever a structural change event
+     * happens. We instead save the current configuration and restore it afterwards.
+     */
+    @Override
+    public void handleLayerEvent(ILayerEvent event) {
+        MutableIntList prevColumnIndexOrder = IntLists.mutable.empty();
```
[the remark this answers is not in the document]
Comment (arnauddieumegard) [reply massactions/plugins/org.polarsys.kitalpha.massactions.core/src/org/polarsys/kitalpha/massactions/core/table/layer/body/MAColumnReorderLayer.java:42]: I don't think so, previously used ArrayList<Integer> is not automatically sorted so the order was kept. MutableIntList have the same behavior (insertion order kept). This prevColumnIndexOrder variable is local and only stores the order.

**Comment by pdulth** [comment]
Hi arnaud, 
We have updated the pipeline library, the api token was not the same for capella and kitalpha
https://github.com/eclipse/capella-pipeline-library/commit/3b6495f8319daa6c06e72ce329a0d160b1fd3416#diff-f9211705ca997c0414d44eddb3f7b0e97d7005e3848c38cb5c36ad94927c3e17R20-R23

**Comment by pdulth** [comment]
current build shall be ok

**Comment by arnauddieumegard** [comment]
> 
> 
> Hi arnaud,
> We have updated the pipeline library, the api token was not the same for capella and kitalpha
> [eclipse/capella-pipeline-library@3b6495f#diff-f9211705ca997c0414d44eddb3f7b0e97d7005e3848c38cb5c36ad94927c3e17R20-R23](https://github.com/eclipse/capella-pipeline-library/commit/3b6495f8319daa6c06e72ce329a0d160b1fd3416#diff-f9211705ca997c0414d44eddb3f7b0e97d7005e3848c38cb5c36ad94927c3e17R20-R23)

Thanks a lot !

**Comment by arnauddieumegard** [comment]
> 
> 
> current build shall be ok

I'm afraid it will not, I think the calls to `github.build***Label(label)` will also fail.

**Comment by bzendagui** [comment]
> > current build shall be ok
> 
> I'm afraid it will not, I think the calls to `github.build***Label(label)` will also fail.

If there are other issues, please revert the commit of the integration of the capella pipeline library. We will integrate it later.

**Comment by arnauddieumegard** [comment]
> 
> 
> > > current build shall be ok
> > 
> > 
> > I'm afraid it will not, I think the calls to `github.build***Label(label)` will also fail.
> 
> If there are other issues, please revert the commit of the integration of the capella pipeline library. We will integrate it later.

It is indeed more complex, it requires to:
- modify the pipeline-library code for label and messages
- publish the PR result on the download.eclipse.org website or link to the product in the ci instance in the generated message

**Comment by pdulth** [comment]
yes, maybe merge it like that, or without the commit with pipeline, in order to have a capella 6.0 with correct nattable and richtext that we can deliver for Team4Capella, then we will think on how we can commonalize all the chain

**Pull request merged**